src/hypothesis/strategies.py

```diff
@@ -711,6 +711,29 @@ def text(
 
 @cacheable
 @defines_strategy
+def text_matching(regex):
```

Comment:
I'm not intrinsically opposed to it being on text, I just think it will create an awful lot of work because you have to deal with things like custom `characters()` strategy.

One question I have (I haven't read through this PR in detail yet due to other work commitments, sorry) is what happens with unicode vs binary regex. *Is* a value from this strategy always text, or could it be binary given a binary regex?